ntbs-integration-tests/Helpers/TestDocumentExtensions.cs

```diff
@@ -1,25 +1,44 @@
-﻿using AngleSharp.Html.Dom;
+﻿using System.Collections.Generic;
+using System.Linq;
+using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
 using Xunit;
 
 namespace ntbs_integration_tests.Helpers
 {
     public static class HtmlDocumentExtensions
     {
-        public static string GetError(this IHtmlDocument document, string input)
+        public static string GetError(this IParentNode document, string input)
         {
-            var errorSpan = document?.QuerySelector($"span[id='{input}-error']");
-            if (errorSpan == null) return null;
+            var errorSpan = document?.QuerySelector($"span#{input}-error");
+            Assert.NotNull(errorSpan);
             if (errorSpan.ClassList.Contains("hidden")) return null;
             return errorSpan.TextContent;
         }
 
-        public static void AssertErrorMessage(this IHtmlDocument document, string inputName, string expectedMessage)
+        public static void AssertErrorMessage(this IParentNode document, string inputName, string expectedMessage)
         {
             var expected = HtmlDocumentHelpers.FullErrorMessage(expectedMessage);
             var actual = document.GetError(inputName);
             Assert.Equal(expected, actual);
         }
 
+        public static void AssertErrorSummaryMessage(this IHtmlDocument document, string summaryInputName, string spanInputName, string expectedMessage)

```

Comment:
It might be quite nice to add documenting comments here tho, to help users of this figure out which bit is what?